### PR TITLE
gex: 0.3.3 -> 0.3.8

### PR DESCRIPTION
**Pull request by Piturnah**
###### Description of changes

[Changelog](https://github.com/Piturnah/gex/blob/master/CHANGELOG.md#038---2023-04-23)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/version-management/gex/default.nix [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
     rev = "v${version}";
-    hash = "sha256-oUcQKpZqqb8wZDpdFfpxLpwdfQlokJE5bsoPwxh+JMM=";
+    hash = "sha256-pjyS0H25wdcexpzZ2vVzGTwDPzyvA9PDgzz81yLGTOY=";
   };
 
-  cargoHash = "sha256-ZFrIlNysjlXI8n78N2Hkff6gAplipxSQXUWG8HJq8fs=";
+  cargoHash = "sha256-+FwXm3QN9bt//dWqzkBzsGigyl1SSY4/P29QtV75V6M=";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl ];
```
Comment (drupol) [inline pkgs/applications/version-management/gex/default.nix:23]: Shouldn't it be in `nativeBuildInputs` as well ?

Comment (Piturnah) [reply pkgs/applications/version-management/gex/default.nix:23]: I'm very new to NixOS so I'm not sure, having trouble even understanding the purpose of `nativeBuildInputs` vs `buildInputs`.

I based this on the inputs for [gitoxide](https://github.com/NixOS/nixpkgs/blob/nixos-23.05/pkgs/applications/version-management/gitoxide/default.nix) which also takes a dependency on openssl, and it allowed the derivation to build fine on my system.

How can I test if it needs to be also in `nativeBuildInputs` for other systems?

Comment (drupol) [reply pkgs/applications/version-management/gex/default.nix:23]: A quick search on Google : https://discourse.nixos.org/t/use-buildinputs-or-nativebuildinputs-for-nix-shell/8464

Try to move openssl in nativebuildinputs and test your derivation to see if it works.

Comment (Piturnah) [reply pkgs/applications/version-management/gex/default.nix:23]: I also found this page, and I don't think `openssl` is used at build time, unless linking against it counts.

I tried moving `openssl` to `nativeBuildInputs` and it now fails to build the derivation.

It works both with `openssl` in `buildInputs` only and with it in both, but as per my first point, I'm not sure that it's necessary for it to be present in both.

**Comment by Azd325** [comment]
Duplicates https://github.com/NixOS/nixpkgs/pull/225549

**Comment by Piturnah** [comment]
> Duplicates #225549

Oops, thank you!

**Pull request closed without merging**